kore/src/Kore/Step.hs

```diff
@@ -146,6 +149,18 @@ data ExecutionMode = All | Any
 type TransitionRule monad rule state =
     Prim -> state -> Strategy.TransitionT rule monad state
 
+profTransitionRule :: forall simplifier
+    . MonadProf simplifier
+    => TransitionRule simplifier
+            (RewriteRule RewritingVariableName)
+            (ProgramState (Pattern RewritingVariableName))
+    -> TransitionRule simplifier
+            (RewriteRule RewritingVariableName)
+            (ProgramState (Pattern RewritingVariableName))
+profTransitionRule rule prim proofState = lift
+    (traceProf ":rewrite:" (runTransitionT (rule prim proofState)))
+        >>= Transition.scatter
+
```

Comment:
Let's also move this to `Kore.Exec` because the function is only used there.

```suggestion
profTransitionRule
    :: forall simplifier
    .  MonadProf simplifier
    => TransitionRule simplifier
            (RewriteRule RewritingVariableName)
            (ProgramState (Pattern RewritingVariableName))
    -> TransitionRule simplifier
            (RewriteRule RewritingVariableName)
            (ProgramState (Pattern RewritingVariableName))
profTransitionRule rule prim proofState = lift
    (traceProf ":rewrite:" (runTransitionT (rule prim proofState)))
        >>= Transition.scatter

```